### PR TITLE
Fix bug with sorting stock states

### PR DESCRIPTION
**Pull request by emord**
https://dimagi-dev.atlassian.net/browse/HI-808

https://sentry.io/organizations/dimagi/issues/1140384958/?project=136860&query=is%3Aunresolved&statsPeriod=14d

### CODE REVIEW

File: corehq/apps/reports/views.py [hunk 1]
```diff
@@ -1023,18 +1023,21 @@ def _get_case_url(case_id):
...
+            product.product_id: product.name
+            for product in SQLProduct.objects.filter(domain=self.domain).values('product_id', 'name').all()
+        }
+
         def _product_name(product_id):
-            try:
-                return SQLProduct.objects.get(product_id=product_id).name
-            except SQLProduct.DoesNotExist:
-                return (_('Unknown Product ("{}")').format(product_id))
+            return product_name_by_id.get(product_id, _('Unknown Product ("{}")').format(product_id))
```
Comment (Rohit25negi) [inline corehq/apps/reports/views.py]: why this functional changed.  i think `SQLProduct.objects.get(product_id=product_id).name` will also give a string name only right?

**Review by Rohit25negi — CHANGES_REQUESTED** [review]
One question left.

File: corehq/apps/reports/views.py (hunk 1 above)
Comment (emord) [reply corehq/apps/reports/views.py]: There should be no functional change here. Before this change there was one query for every entry in the map. Now there is only one query to get all names and product ids. If there product_id does not exist, then we return "unknown product" if it does exist we return the name from `SQLProduct`

Comment (Rohit25negi) [reply corehq/apps/reports/views.py]: COol.

**Review by Rohit25negi — APPROVED** [review]

**Pull request merged**